firefly_client/firefly_client.py

```diff
@@ -199,30 +225,39 @@ def __init__(self, url, channel, html_file=_def_html_file, token=None):
         self.session = requests.Session()
         token and ssl and self.session.headers.update(self.auth_headers)
         not ssl and token and warn('token ignored: should be None when url starts with http://')
+        if viewer_override:
+            self.firefly_viewer = viewer_override
+        else:
+            self.firefly_viewer = FireflyClient.SLATE_VIEWER if html_file == 'slate.html' \
+                else FireflyClient.TRIVIEW_VIEWER
         debug('new instance: %s' % url)
 
-    def _lab_env_tab_start(self, tab_type):
+    def _lab_env_tab_start(self, tab_type, html_file):
         """start a tab in the lab environment, tab_type must be 'lab' or 'browser' """
         self.lab_env_tab_type = tab_type
         if tab_type == BROWSER:
             idx = self.channel.find('__viewer')
             c = self.channel[0:idx] if idx > -1 else self.channel  # the ext will add '__viewer' so I have to remove it
-            self.dispatch(ACTION_DICT['StartBrowserTab'], {'channel': c}, Env.firefly_channel_lab)
+            self.dispatch(ACTION_DICT['StartBrowserTab'],
+                          {'channel': c, 'fireflyHtmlFile': _def_html_file}, Env.firefly_channel_lab)
         elif tab_type == LAB:
             if not self.render_tree_id:
-                self.render_tree_id = 'slateClient-%s-%s' % (len(self.instances), round(time.time()))
-            self.show_lab_tab()
+                self.render_tree_id = 'slate-1'  # no longer generating redner_tree_id
```

Comment:
1. I don't think we should refer to this as "slate" anymore because slate is just one of the two modes. It can be something like "client_viewer"
2. Are we certain that the user always wants to open one client for different notebooks? It's beneficial to combine results from different notebooks in one triview but I can also think of a case that 2 notebooks were two different contexts and they wanted to separate that work in two different clients. We might wanna take scientists/user opinion before removing this capabilty of more than one client.

Reply:
> I don't think we should refer to this as "slate" anymore because slate is just one of the two modes. It can be something like "client_viewer"

I agree

> Are we certain that the user always wants to open one client for different notebooks? 

Just to review the conversation @jaladh-singhal and I had at the stand up

- The capability of the jupyter lab extension opening multiple tabs (in JL) has been very buggy and it requiring a lot of work for a corner case
    - This case is when you use `make_lab_client()`
    - The solution is that  `make_lab_client()` open a singleton.
- The `make_client()` function which opens browsers tabs does not have a limit. The user can open as many as he wants.
- This still give the use the capability without us having support all the corner cases